src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -1438,37 +1385,31 @@ void c_typecheck_baset::typecheck_expr_rel_vector(
 
 void c_typecheck_baset::typecheck_expr_ptrmember(exprt &expr)
 {
-  if(expr.operands().size()!=1)
-  {
-    error().source_location = expr.source_location();
-    error() << "ptrmember operator expects one operand" << eom;
-    throw 0;
-  }
-
-  const typet &op0_type = expr.op0().type();
+  auto &op = to_unary_expr(expr).op();
```

Comment:
There should be a comment here explaining that the method does not take a `unary_exprt`, because it may turn the input expression into a binary expression.